src/loader/gtfs/trip.cc

```diff
@@ -242,28 +242,12 @@ std::string trip::display_name(timetable const& tt) const {
                        utl::parse<int>(short_name_));
   }
 
-  // take the first thing that starts with a letter and ends with a number
-  auto const starts_with_letter_and_ends_with_number =
-      [=](std::string_view line_id) {
-        return !is_digit(line_id.front()) && is_digit(line_id.back());
-      };
+  auto precedence = std::array{std::string_view{route_->short_name_},
```

Comment:
This doesn't work for all feeds that use semantic naming instead of the google maps convention.
That is as far as I know most of the data for Germany and Austria

Reply:
as far as I remember for example the ÖBB feed entirely depends on this starts-with-letter-ends-with-number heuristic to display the right string

Reply:
If there are only very few feeds with this problem we could also try to contact them and ask them if it's possible to write something useful into `route_name_short`. I would see Google as the inventor of GTFS as kind of a aid in interpreting the standard and point them to the Google way of doing things as "this is how it's done".